### PR TITLE
Do not count deleted bundledeployments in cluster status

### PR DESCRIPTION
**Pull request by manno**
Refers to https://github.com/rancher/fleet/issues/2448

### CODE REVIEW

File: internal/cmd/controller/reconciler/cluster_controller.go [hunk 1]
```diff
@@ -94,15 +94,20 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
...
 	for _, bundle := range cleanup {
-		for _, bundleDeployment := range bundleDeployments.Items {
-			if bundleDeployment.Labels[fleet.BundleLabel] == bundle.Name &&
-				bundleDeployment.Labels[fleet.BundleNamespaceLabel] == bundle.Namespace {
-				logger.V(1).Info("cleaning up bundleDeployment not matching the cluster", "bundledeployment", bundleDeployment)
-				err := r.Delete(ctx, &bundleDeployment) // nolint:gosec // does not store pointer
+		for _, bd := range bundleDeployments.Items {
+			if deleted[bd.GetUID()] {
+				continue
+			}
```
Comment (0xavi0) [inline internal/cmd/controller/reconciler/cluster_controller.go:103]: Question: when is this happening? 
is it possible to have a `bundledeployment` that matches the condition to be deleted:
```go
bd.Labels[fleet.BundleLabel] == bundle.Name && bd.Labels[fleet.BundleNamespaceLabel] == bundle.Namespace
```
more than once?

File: internal/cmd/controller/reconciler/cluster_controller.go [hunk 2]
```diff
@@ -94,15 +94,20 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
...
+			if bd.Labels[fleet.BundleLabel] == bundle.Name && bd.Labels[fleet.BundleNamespaceLabel] == bundle.Namespace {
+				logger.V(1).Info("cleaning up bundleDeployment not matching the cluster", "bundledeployment", bd)
+				err := r.Delete(ctx, &bd) // nolint:gosec // does not store pointer
```
Comment (0xavi0) [inline internal/cmd/controller/reconciler/cluster_controller.go:106]: Nit: I THINK we don't need this `// nolint` statement?. (I known it's not part of the changes of the PR, but just saw it and remembered that I've seen it in other files) 
I've tested running `golangci-lint` locally without the `// nolint` and it didn't complain.

File: internal/cmd/controller/reconciler/cluster_controller.go (hunk 1 above)
Comment (manno) [reply internal/cmd/controller/reconciler/cluster_controller.go:103]: Ah, I just wanted to skip the whole inner loop. But you're right, the existing `if` condition is probably not  significantly slower than `deleted[bd.GetUID()]`.

File: internal/cmd/controller/reconciler/cluster_controller.go (hunk 2 above)
Comment (manno) [reply internal/cmd/controller/reconciler/cluster_controller.go:106]: Ah, nice. Last time I checked I still got warnings. I'll add a commit, but leave the ones in gitrepo controller to you, so we don't conflict.

**Review by 0xavi0 — APPROVED** [review]
LGTM

**Pull request merged**